src/internal/_functionName.js

```diff
@@ -0,0 +1,5 @@
+module.exports = function _functionName(f) {
+  // String(x => x) evaluates to "x => x", so the pattern may not match.
+  var match = String(f).match(/^function (\w*)/);
+  return match == null ? '' : match[1];
```

Comment:
> in that case you have dragged all platforms down to IE's level.

It sounds as though you *are* concerned about the performance implications. ;)

[MDN][1] states the following:

> This is a new technology, part of the ECMAScript 2015 (ES6) standard.

It seems the approach above is the "correct" approach, though it's undoubtedly ugly.


[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name